numba/cpython/setobj.py

```diff
@@ -1005,11 +1008,13 @@ def _allocate_payload(self, nentries, realloc=False):
         with builder.if_then(builder.load(ok), likely=True):
             if realloc:
                 meminfo = self._set.meminfo
-                ptr = context.nrt.meminfo_varsize_alloc(builder, meminfo,
-                                                        size=allocsize)
+                ptr = context.nrt.meminfo_varsize_realloc(builder, meminfo,
```

Comment:
It doesn't seem like alloc_ok is used in either path. It seems like the contents of the builder.if_else should be replaced with `alloc_ok`.